### PR TITLE
[SWAT-397] Annotations aspect framework

### PR DESCRIPTION
**Pull request by gjorquera**
The acceptance spec contains the different use cases that this framework supports.

### CODE REVIEW

File: spec/acceptance/mehtod_annotations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
```
Comment (warseph) [inline spec/acceptance/mehtod_annotations_spec.rb]: typo on the filename :wink:

File: lib/co_aspects/attacher.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        @pending.each do |pending|
+          pending.aspect.apply klass,
+            method: method_name,
+            args: pending.args,
+            block: pending.block
+        end
+        @enabled = true
+        @pending = []
+      end
+    end
```
Comment (warseph) [inline lib/co_aspects/attacher.rb:24]: What do you think about adding a private method to handle the blocking? Something like:

``` ruby
def attach(klass, method_name)
  blocking do
    @pending.each do |pending|
      pending.aspect.apply klass,
        method: method_name,
        args: pending.args,
        block: pending.block
    end
    @pending = []
  end
end

private

def blocking
  @enabled = true unless defined?(@enabled)
  if @enabled
    @enabled = false
    yield
    @enabled = true
  end
end
```

This way you can leave the actual logic a bit more clear (and you could remove the `@enabled = true` from the initializer)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  end
+end
+```
+
+The aspect attached to a method is the camelized version of the annotation
+followed by "Aspect". For example, the `_rescue_and_notify` will attach the
+`RescueAndNotifyAspect` aspect.
+
+Several aspects can be attached to the same method, and the same aspect can be
+attached to different methods.
```
Comment (warseph) [inline README.md:48]: Maybe add a bit of info regarding Aspect parameters?

File: spec/acceptance/mehtod_annotations_spec.rb (hunk 1 above)
Comment (gjorquera) [inline spec/acceptance/mehtod_annotations_spec.rb]: Where? :confounded:

File: lib/co_aspects/attacher.rb (hunk 2 above)
Comment (gjorquera) [inline lib/co_aspects/attacher.rb:24]: I like it!

File: README.md (hunk 3 above)
Comment (gjorquera) [inline README.md:48]: Sure.

**Pull request merged**